src/ferny/errors.py

```diff
@@ -90,7 +94,8 @@ def make_gaierror_map() -> Iterable[Tuple[str, int]]:
 
 
 def get_exception_for_ssh_stderr(stderr: str) -> Exception:
-    for ssh_cls in [AuthenticationError, HostKeyError]:
+    # ChangedHostKeyError must come first, as the HostKeyError message is also present in the "key has changed" error
+    for ssh_cls in [AuthenticationError, ChangedHostKeyError, HostKeyError]:
```

Comment:
Could we change the other message so that the conditions are not overlapping and the order is irrelevant?  Otherwise I'd prefer something slightly more formal for ordering ­— maybe lifting this out to a top-level constant for example.  I completely missed the comment here on first read.

Reply:
For "changed" it looks like this:
```
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the RSA key sent by the remote host is
SHA256:ZeBykpPEufJoNUuDWQTEnA+pIdkQUralwtLLItxK+GM.
Please contact your system administrator.
Add correct host key in /tmp/pytest-of-martin/pytest-103/test_no_host_key_changed0/keys/known_hosts to get rid of this message.
Offending RSA key in /tmp/pytest-of-martin/pytest-103/test_no_host_key_changed0/keys/known_hosts:1
Host key for [127.0.0.1]:53187 has changed and you have requested strict checking.
Host key verification failed.
```

For "unknown" it looks like this:
```
No RSA host key is known for [127.0.0.1]:51609 and you have requested strict checking.
Host key verification failed.
```
So the "No .* key is known" would provide a non-overlapping match. We could try to match on the entire message instead of just a single line, and include the whole thing in the error message.

Reply:
Done, and I split off `UnknownHostKey` to make this more symmetric and safer to use.

Reply:
Ah dang, [RHEL 8 can't do that](https://github.com/allisonkarlitskaya/ferny/actions/runs/5088646507/jobs/9145312889?pr=15). I'll need to rework that using conditional xfails, and a RHEL 8 specific test perhaps.

Reply:
I reworked this to keep the generic error as a fallback. Unfortunately this means to bring back the order dependence, I updated the comment.